management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -740,39 +737,46 @@ public void clearSnapshots(
       @RpcParam(name = "full") Boolean full,
       @RpcParam(name = "notifications") boolean notifications,
-      @RpcParam(name = "repairParallelism") Optional<RepairParallelism> repairParallelism,
-      @RpcParam(name = "datacenters") Optional<Collection<String>> datacenters,
-      @RpcParam(name = "associatedTokens") Optional<List<RingRange>> associatedTokens,
-      @RpcParam(name = "repairThreadCount") Optional<Integer> repairThreadCount)
+      @RpcParam(name = "repairParallelism") String repairParallelism,
+      @RpcParam(name = "datacenters") List<String> datacenters,
+      @RpcParam(name = "associatedTokens") String ringRangeString,
+      @RpcParam(name = "repairThreadCount") int repairThreadCount)
       throws IOException {
     // At least one keyspace is required
     assert (keyspace != null);
     Map<String, String> repairSpec = new HashMap<>();
-    repairParallelism.map(rPar -> repairSpec.put(RepairOption.PARALLELISM_KEY, rPar.getName()));
+    // add tables/column families
+    if (tables != null && !tables.isEmpty()) {
+      repairSpec.put(RepairOption.COLUMNFAMILIES_KEY, String.join(",", tables));
+    }
+    // set incremental reapir
     repairSpec.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(!full));
-    repairThreadCount.map(
-        tCount ->
-            repairSpec.put(
-                RepairOption.JOB_THREADS_KEY, Integer.toString(tCount == 0 ? 1 : tCount)));
+    // Parallelism should be set if it's requested OR if incremental repair is requested.
+    if (!full) {
+      // Incremental repair requested, make sure parallelism is correct
+      if (repairParallelism != null
+          && !RepairParallelism.PARALLEL.getName().equals(repairParallelism)) {
+        throw new IOException(
+            "Invalid repair combination. Incremental repair if Parallelism is not set");
+      }
+      // Incremental repair and parallelism should be set
+      repairSpec.put(RepairOption.PARALLELISM_KEY, RepairParallelism.PARALLEL.getName());
+    }
+    // thread count should be at least 1
+    repairSpec.put(
+        RepairOption.JOB_THREADS_KEY, Integer.toString(Integer.max(1, repairThreadCount)));
     repairSpec.put(RepairOption.TRACE_KEY, Boolean.toString(Boolean.FALSE));
-    tables.map(
-        tabs -> repairSpec.put(RepairOption.COLUMNFAMILIES_KEY, StringUtils.join(tables, ",")));
-    if (full) {
-      associatedTokens.map(
-          aTokens ->
-              repairSpec.put(
-                  RepairOption.RANGES_KEY,
-                  StringUtils.join(
-                      aTokens.stream()
-                          .map(token -> token.getStart() + ":" + token.getEnd())
-                          .collect(Collectors.toList()),
-                      ",")));
-    }
-    datacenters.map(
-        dcs -> repairSpec.put(RepairOption.DATACENTERS_KEY, StringUtils.join(dcs, ",")));
+    // add ranges if reapir is full
+    if (full && ringRangeString != null && !ringRangeString.isEmpty()) {
```

Comment:
Suggestion: I'd be more inclined to throw an error here if `!full && ringRangeString != null`, since it indicates an invalid request AFAICT?

Reply:
I'm not sure! Based on your logic here it seemed that you only wanted to add the `RANGES_KEY` when the repair type was `full`. So I assumed that ranges was only valid for full repairs. 

In my original logic, I think we just don't add the ranges if they aren't specified, which might be the the best way to do it. Let Cassandra throw an error if there's an error to be thrown?

Reply:
Ok, I'll investigate this and update accordingly.

Reply:
Awesome!